src/app/admin/createUser/onChangetextDeferredError.ts

```diff
@@ -0,0 +1,29 @@
+import { Errors, Fields, FormErrors, getErrorType, Setter } from "@/components/Form/formFunctions";
+import { SelectChangeEventHandler } from "@/components/DataInput/inputHandlerFactories";
+import { User } from "@supabase/gotrue-js";
+
+const onChangeTextDeferredError = <SpecificFields extends Fields>(
+    fieldSetter: Setter<SpecificFields>,
+    errorSetter: Setter<FormErrors<SpecificFields>> | Setter<Required<FormErrors<SpecificFields>>>,
+    key: keyof SpecificFields,
+    required?: boolean,
+    regex?: RegExp,
+    InvitedUserSetter?: { (user: User | null): void; (arg0: null): void } | undefined,
+    formattingFunction?: (value: string) => SpecificFields[keyof SpecificFields],
+    additionalCondition?: (value: string) => boolean
+): SelectChangeEventHandler => {
+    return (event) => {
+        InvitedUserSetter?.(null);
```

Comment:
This is the only part which is not generic, but if we rename the prop to something like `additionalAction?: () => void` then pass in the "clear invited user" as a void function we can keep this generic.